src/coreclr/src/inc/corprof.idl

```diff
@@ -769,6 +774,14 @@ typedef enum
     COR_PRF_EVENTPIPE_VERBOSE = 5
 } COR_PRF_EVENTPIPE_LEVEL;
 
+typedef struct
+{
+    const WCHAR* providerName;
+    UINT64       keywords;
+    UINT32       loggingLevel;
+    const WCHAR* filterData;
```

Comment:
From what I can see the format is "key=value;key2=value2;" with optional quotes to escape '=' and ';'. I added a comment to corprof.idl to indicate this.

Reply:
I haven't looked into the implementation required to change it, but just wanted to raise that as a brand new interface we have the opportunity to do so. My guess is that folks using the API wouldn't say escaped string encodings are ideal, but also not a huge deal that blocks their progress. So entirely your call if the implementation cost looks like it would be worth making a change. My recollection is that ETW uses a different encoding where items are separated by embedded null characters with a double-null to indicate the end - no escape characters needed. Or an object model style API could use an array of structs with each struct having key and value fields.